From a8fa6c930597b633367777044f4c0328012f6bd5 Mon Sep 17 00:00:00 2001
From: Fabian Kosmale <fabian.kosmale@qt.io>
Date: Tue, 1 Jun 2021 16:40:44 +0200
Subject: [PATCH 13/19] Make QaccessibleQuickWidget private API

Its base class is private API, so it should be private API, too.

Change-Id: Ic80f841fee19ed0305c60ad5f8e9349a05f09e5e
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
Reviewed-by: Ulf Hermann <ulf.hermann@qt.io>
Reviewed-by: Qt CI Bot <qt_ci_bot@qt-project.org>
(cherry picked from commit a4fa74d3e7581cb5c6bb82223ee17257f66fa41d)
---
 src/quickwidgets/qaccessiblequickwidget.cpp           |  2 +-
 ...ssiblequickwidget.h => qaccessiblequickwidget_p.h} | 11 +++++++++++
 src/quickwidgets/qaccessiblequickwidgetfactory.cpp    |  2 +-
 src/quickwidgets/quickwidgets.pro                     |  2 +-
 4 files changed, 14 insertions(+), 3 deletions(-)
 rename src/quickwidgets/{qaccessiblequickwidget.h => qaccessiblequickwidget_p.h} (92%)

diff --git a/src/quickwidgets/qaccessiblequickwidget.cpp b/src/quickwidgets/qaccessiblequickwidget.cpp
index 6f04d6693f..8a1c901880 100644
--- a/src/quickwidgets/qaccessiblequickwidget.cpp
+++ b/src/quickwidgets/qaccessiblequickwidget.cpp
@@ -37,7 +37,7 @@
 **
 ****************************************************************************/
 
-#include "qaccessiblequickwidget.h"
+#include "qaccessiblequickwidget_p.h"
 
 #include "qquickwidget_p.h"
 
diff --git a/src/quickwidgets/qaccessiblequickwidget.h b/src/quickwidgets/qaccessiblequickwidget_p.h
similarity index 92%
rename from src/quickwidgets/qaccessiblequickwidget.h
rename to src/quickwidgets/qaccessiblequickwidget_p.h
index 1f52c78c46..7c2ab930e0 100644
--- a/src/quickwidgets/qaccessiblequickwidget.h
+++ b/src/quickwidgets/qaccessiblequickwidget_p.h
@@ -40,6 +40,17 @@
 #ifndef QACCESSIBLEQUICKWIDGET_H
 #define QACCESSIBLEQUICKWIDGET_H
 
+//
+//  W A R N I N G
+//  -------------
+//
+// This file is not part of the Qt API.  It exists purely as an
+// implementation detail.  This header file may change from version to
+// version without notice, or even be removed.
+//
+// We mean it.
+//
+
 #include "qquickwidget.h"
 #include <QtWidgets/qaccessiblewidget.h>
 
diff --git a/src/quickwidgets/qaccessiblequickwidgetfactory.cpp b/src/quickwidgets/qaccessiblequickwidgetfactory.cpp
index 3756d0c27c..7ba88a1769 100644
--- a/src/quickwidgets/qaccessiblequickwidgetfactory.cpp
+++ b/src/quickwidgets/qaccessiblequickwidgetfactory.cpp
@@ -38,7 +38,7 @@
 ****************************************************************************/
 
 #include "qaccessiblequickwidgetfactory_p.h"
-#include "qaccessiblequickwidget.h"
+#include "qaccessiblequickwidget_p.h"
 
 QT_BEGIN_NAMESPACE
 
diff --git a/src/quickwidgets/quickwidgets.pro b/src/quickwidgets/quickwidgets.pro
index f46deb54ac..85d156b8a3 100644
--- a/src/quickwidgets/quickwidgets.pro
+++ b/src/quickwidgets/quickwidgets.pro
@@ -8,7 +8,7 @@ HEADERS += \
     qquickwidget.h \
     qquickwidget_p.h \
     qtquickwidgetsglobal.h \
-    qaccessiblequickwidget.h \
+    qaccessiblequickwidget_p.h \
     qaccessiblequickwidgetfactory_p.h
 
 SOURCES += \
-- 
2.40.0

